ludwig/backend/ray.py

```diff
@@ -407,8 +452,34 @@ def to_tensors(df: pd.DataFrame) -> pd.DataFrame:
     def predict_single(self, batch):
         raise NotImplementedError("predict_single can only be called on a local predictor")
 
-    def batch_evaluation(self, dataset, collect_predictions=False, **kwargs):
-        raise NotImplementedError("Ray backend does not support batch evaluation at this time.")
+    def batch_evaluation(self, dataset: RayDataset, collect_predictions: bool = False, **kwargs):
+        runner = Trainer(**{**get_trainer_kwargs(), **self.trainer_kwargs})
+        runner.start()
+        try:
+            datasets = {"eval": dataset.pipeline(shuffle=False, **self.data_loader_kwargs)}
+            predictor_kwargs = {
+                **self.predictor_kwargs,
+                "collect_predictions": False,
```

Comment:
Not 100% sure why we set collect_predictions to False here, but later (480), it it was true, then we run inference again to collect them. What's the rationale?

Reply:
We need to be a Horovod context to collect the aggregated metrics, since it relies on collective communication ops. However, Horovod is not suitable for transforming one big dataset to another. For that you need something like Dask or Ray Datasets. So we need to break these up into two different steps. I can add a comment to this effect.

Reply:
Got it, this may make the process quite slower when collect predictions is true I imagine, but that's fine, if the batch predict at line 480 as runs in parallel, this will already be faster with 3 machines than it would be with one.

Probably a clarifying comment here and around 480 would be useful as other people encountering these may thing it's a redundant thing.

Reply:
Comment added. Yeah, will be interesting to see how much of penalty it imposes to break these steps up. One nice thing about Ray Datasets is we can typically cache the whole dataset in the cluster, so two passes over the data ends up being a lot less expensive than going to remote files each time. But still would be nice to find a way to do it in one go.